File: 5.zero-message-queue/data-server/main.py
import zmq
import json

datastore = {}

context = zmq.Context()
sock = context.socket(zmq.REP)
sock.bind("tcp://*:27017")

while True:
    req = sock.recv_json()
    print("Received request: %s" % req)

    if req['type'] == 'set':
        datastore[req['id']] = req['value']
        sock.send_json({ 'error': None })
    elif req['type'] == 'get':
        sock.send_json({ 'ret': datastore, 'error': None })
    else:
        sock.send_json({ 'error': '未知请求！' })

File: 5.zero-message-queue/data-server/main.js
const zmq = require('zeromq')
var datastore = {
}

var main = async () => {
    var sock = zmq.socket('rep')
    sock.bindSync('tcp://*:27017')

    sock.on('message', async (msg) => { 
        var req = JSON.parse(msg)
        console.log(req)
        switch (req.type) {
            case 'set':
                datastore[req.id] = req.value;
                sock.send(JSON.stringify({ error: null }))
                break;
            case 'get':
                sock.send(JSON.stringify({ ret: datastore, error: null }))
                break;
            default:
                sock.send(JSON.stringify({ error: '未知请求！' }))
                break;
        }
    })
}

main()